### PR TITLE
docs(faq): debug apisix with gdb

### PR DESCRIPTION
**Pull request by shuaijinchao**
### Description
Add debug apisix with gdb

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/FAQ.md [hunk 1]
```diff
@@ -566,3 +566,72 @@ The difference between `plugin-metadata` and `plugin-configs`:
...
+```bash
+$ tail -f  logs/error.log
+[alert] 29768#0: worker process 29770 exited on signal 11 (core dumped)
+````
+
+You can enable `Core Dumps` to use `GDB` to locate the specific cause of the problem.
+
+Proceed as follows:
+
+### Enable APISIX CoreDumps option
```
Comment (moonming) [inline docs/en/latest/FAQ.md:583]: `CoreDumps` -> `core dump`

File: docs/en/latest/FAQ.md [hunk 2]
```diff
@@ -566,3 +566,72 @@ The difference between `plugin-metadata` and `plugin-configs`:
...
+
+APISIX enables CoreDumps related options by default, no additional configuration is required.
```
Comment (moonming) [inline docs/en/latest/FAQ.md:585]: If it is enabled by default, then this line of text is not needed?

File: docs/en/latest/FAQ.md [hunk 3]
```diff
@@ -566,3 +566,72 @@ The difference between `plugin-metadata` and `plugin-configs`:
...
+### Check System CoreDumps Limit
+
+```bash
+$ ulimit -c
+unlimited
+# If 0, use the following commands to lift the limit or set a custom size.
+$ ulimit -c unlimited
+```
+
+### Setting CoreDumps save path and filename (optional)
```
Comment (moonming) [inline docs/en/latest/FAQ.md:596]: `Enabling core dump in the Operating System`

File: docs/en/latest/FAQ.md [hunk 4]
```diff
@@ -566,3 +566,72 @@ The difference between `plugin-metadata` and `plugin-configs`:
...
+
+Proceed as follows:
+
+### Enable APISIX CoreDumps option
+
+APISIX enables CoreDumps related options by default, no additional configuration is required.
+
+### Check System CoreDumps Limit
+
+```bash
```
Comment (moonming) [inline docs/en/latest/FAQ.md:589]: why not `ulimit -c unlimited` directly?

File: docs/en/latest/FAQ.md [hunk 5]
```diff
@@ -566,3 +566,72 @@ The difference between `plugin-metadata` and `plugin-configs`:
...
+unlimited
+# If 0, use the following commands to lift the limit or set a custom size.
+$ ulimit -c unlimited
+```
+
+### Setting CoreDumps save path and filename (optional)
+
+```bash
+$ sysctl -w kernel.core_pattern=/tmp/core-%e.%p.%t
+$ sysctl -p /etc/syslog.conf
```
Comment (moonming) [inline docs/en/latest/FAQ.md:600]: I think you need `sudo`

Comment (moonming) [reply docs/en/latest/FAQ.md:600]: And for CentOS and FreeBSD, the cmds are different.

File: docs/en/latest/FAQ.md [hunk 6]
```diff
@@ -566,3 +566,72 @@ The difference between `plugin-metadata` and `plugin-configs`:
...
+
+```bash
+$ ll /tmp
+ rw------- 1  root  root  280 MiB  Mon Mar 21 14:11:22 2022  core-openresty.29770.1647843081
+```
+
+### Debug and view the call stack with GDB, For example:
+
+```bash
+$ gdb /usr/local/openresty/bin/openresty /tmp/core-openresty.29770.1647843081
```
Comment (moonming) [inline docs/en/latest/FAQ.md:613]: do you need `sudo`?

Comment (moonming) [reply docs/en/latest/FAQ.md:613]: Why the path of openresty is fixed?

File: docs/en/latest/FAQ.md [hunk 7]
```diff
@@ -566,3 +566,72 @@ The difference between `plugin-metadata` and `plugin-configs`:
...
+#13 0x0000562957ee8634 in ngx_http_process_request_line (rev=0x7f78fcf3ba90) at src/http/ngx_http_request.c:1154
+#14 0x0000562957ed0026 in ngx_epoll_process_events (cycle=<optimized out>, timer=<optimized out>, flags=<optimized out>) at src/event/modules/ngx_epoll_module.c:901
+#15 0x0000562957ec68da in ngx_process_events_and_timers (cycle=cycle@entry=0x5629595b78f0) at src/event/ngx_event.c:257
+#16 0x0000562957ece370 in ngx_worker_process_cycle (cycle=cycle@entry=0x5629595b78f0, data=data@entry=0x1) at src/os/unix/ngx_process_cycle.c:811
+#17 0x0000562957ecc85d in ngx_spawn_process (cycle=cycle@entry=0x5629595b78f0, proc=proc@entry=0x562957ece2f0 <ngx_worker_process_cycle>, data=data@entry=0x1,
+    name=name@entry=0x562958023839 "worker process", respawn=respawn@entry=-3) at src/os/unix/ngx_process.c:199
+#18 0x0000562957ecda2c in ngx_start_worker_processes (cycle=cycle@entry=0x5629595b78f0, n=2, type=type@entry=-3) at src/os/unix/ngx_process_cycle.c:387
+#19 0x0000562957eceea2 in ngx_master_process_cycle (cycle=0x5629595b78f0) at src/os/unix/ngx_process_cycle.c:135
+#20 0x0000562957ea56c0 in main (argc=<optimized out>, argv=<optimized out>) at src/core/nginx.c:385
+```
```
Comment (moonming) [inline docs/en/latest/FAQ.md:637]: why we need this?

**Review by spacewander — CHANGES_REQUESTED** [review]
Is this necessary? We can simply find tens of gdb tutorials about how to use gdb...

**Comment by moonming** [comment]
> Is this necessary? We can simply find tens of gdb tutorials about how to use gdb...

yes, I agreed with you

**Comment by shuaijinchao** [comment]
If you all agree that this FAQ is not necessary, I will close this PR.

**Pull request closed without merging**